xinference/model/utils.py

```diff
@@ -102,3 +103,22 @@ def retry_download(
             raise RuntimeError(
                 f"Failed to download model '{model_name}' after multiple retries"
             )
+
+
+def valid_model_revision(
+    meta_path: str, expected_model_revision: Optional[str]
+) -> bool:
+    if not os.path.exists(meta_path):
```

Comment:
Address legacy meta files to ensure that upgrading xinference does not require users to download the existing model again.

To accomplish this, verify if the meta file content is in JSON format. If it's not in JSON format, this indicates that the meta file is legacy and should return True directly. Otherwise, check the model revision.

Log an entry when legacy meta files are detected and include a unit test.